pkg/controller/elasticsearch/pdb/dfs.go

```diff
@@ -0,0 +1,153 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package pdb
+
+import (
+	"slices"
+
+	appsv1 "k8s.io/api/apps/v1"
+
+	esv1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/elasticsearch/sset"
+)
+
+var (
+	dataRoles = []string{
+		string(esv1.DataRole),
+		string(esv1.DataHotRole),
+		string(esv1.DataWarmRole),
+		string(esv1.DataColdRole),
+		string(esv1.DataContentRole),
+		// Note: DataFrozenRole is excluded as it has different disruption rules (yellow+ health)
+	}
+)
+
+// normalizeRole returns the normalized form of a role where any data role
+// is normalized to the same data role.
+func normalizeRole(role string) string {
+	if slices.Contains(dataRoles, role) {
+		return string(esv1.DataRole)
```

Comment:
I think we should only do that if the user has specified the generic data role. If they use the tiered data roles we should keep them and created dedicated PDBs. The rationale being that with tiered data roles indices are very likely to be fully within one of the tiers. For example for the the hot tier both primary and replica shards are in the hot tier. So the tier should be considered as an logical availability domain and be protected via a dedicated PDB. If a a node in the hot tier is disrupted and we are in yellow health then further disruptions in let's say the cold data tier won't make the situation worse.